### DISCUSSION TITLE
Support Annotated annotations on non-field parameters (Request, BackgroundTasks, etc.)

### DISCUSSION THREAD

**Discussion by mjpieters** [fastapi/fastapi#9466] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, BackgroundTasks
from thirdparty_library import decorator, MarkerAnnotation

api = FastAPI()

@api.route("/foo")
@decorator
def foo_endpoint(bar: int, background_tasks: Annotated[BackgroundTasks, MarkerAnnotation]):
    ...
```

### Description

I'm helping out with the design of a third-party FastAPI add-on, and I'm proposing the project uses [`typing.Annotated`](https://docs.python.org/3/library/typing.html#typing.Annotated) to attach metadata to argument types, just like FastAPI supports using `Depends()` annotations and pydantic supports `Field` annotations.

However, if you try to use `Annotated[...]` on any of the non-field parameter dependencies (`Request`, `WebSocket`, `HTTPConnection`, `Response`, `BackgroundTasks` and `SecurityScopes`) then injection of those dependencies breaks.

That's because the code to introspect the endpoint uses `inspect.signature()` (with additional forwarded reference resolving) and then looks at each of the the `inspect.Parameter.annotation` values for the endpoint arguments directly, using `pydantic.utils.lenient_issubclass()`:

https://github.com/tiangolo/fastapi/blob/8ac8d70d52bb0dd9eb55ba4e22d3e383943da05c/fastapi/dependencies/utils.py#L335-L356

When you use `Annotated[BackgroundTasks, ...]` in your endpoint signature then `lenient_issubclass()` will return `False`.

Because `Annotated` is always flattened, you can reliably unwrap it by testing the origin first and then taking the first argument:

```python
# Unwrap Annotated annotations
annotation = param.annotation
if typing.get_origin(annotation) is typing.Annotated:
    annotation = typing.get_args(annotation)[0]
```

Could FastAPI please do so here? That way third-party libraries can add their own enhancements to endpoint signatures.

### Operating System

macOS

### FastAPI Version

0.92.0

### Python Version

3.11.3

**Comment by Kludex** [comment]
Interesting idea. 👍 

I've added the label `investigate` for @tiangolo to check later. 🙏

**Comment by YuriiMotov** [comment]
Works in current version:

```python
from typing import Annotated

from fastapi import FastAPI, BackgroundTasks, Request
from fastapi.testclient import TestClient


class MarkerAnnotation:
    pass

app = FastAPI()

@app.get("/foo")
def foo_endpoint(
    background_tasks: Annotated[BackgroundTasks, MarkerAnnotation()],
    request: Annotated[Request, MarkerAnnotation()],
):
    assert isinstance(background_tasks, BackgroundTasks)
    assert isinstance(request, Request)
    return "ok"


def test():
    client = TestClient(app)

    resp = client.get("/foo")
    assert resp.status_code == 200
```
